packages/components/src/components/modal/modal-styles.ts

```diff
@@ -128,41 +142,44 @@ export const getComponentCss = (
         boxSizing: 'border-box',
         transition: `transform .6s ${transitionTimingFunction}`,
         transform: open ? 'scale3d(1,1,1)' : 'scale3d(.9,.9,1)',
-        padding: pxToRemWithUnit(32),
+        padding: !disableCloseButton ? `${pxToRemWithUnit(32)} 32px 32px 32px` : '32px', // rem value needed to prevent overlapping of close button and contents in scaling mode
         backgroundColor: lightThemeBackgroundColor,
-        ...getFocusVisibleFallback(getFocusJssStyle({ color: lightThemeBackgroundColor })),
-        [mediaQueryM]: {
-          padding: pxToRemWithUnit(40),
+        outline: 0,
+        '&:focus::before': {
+          content: '""',
+          position: 'fixed',
+          border: `${borderWidthBase} solid`,
+          ...buildResponsiveStyles(fullscreen, (fullscreenValue: boolean) => ({
+            borderRadius: fullscreenValue ? 0 : '12px',
+            borderColor: fullscreenValue ? lightThemePrimaryColor : darkThemePrimaryColor,
+            ...(fullscreenValue ? getInsetJssStyle(0) : getInsetJssStyle(-4)),
```

Comment:
Could move condition next to function parameter like `getInsetJssStyle(fullscreenValue ? 0 : -4)`.